bundler/lib/bundler/cli/outdated.rb

```diff
@@ -98,27 +100,24 @@ def run
       end
 
       if outdated_gems.empty?
-        unless options[:parseable]
+        if options[:json]
+          print_gems_json([])
+        elsif !options[:parseable]
           Bundler.ui.info(nothing_outdated_message)
         end
       else
-        if options_include_groups
-          relevant_outdated_gems = outdated_gems.group_by {|g| g[:groups] }.sort.flat_map do |groups, gems|
-            contains_group = groups.split(", ").include?(options[:group])
-            next unless options[:groups] || contains_group
```

Comment:
Note that this incidentally fixes a bug though probably nobody is bothered by it - in current master, if you supply both `--groups` and `--group`, the `--group` option will not be used to filter the results as expected, because of the short-circuit here.